core/state/statedb_logger.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2024 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package state
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/stateless"
+	"github.com/ethereum/go-ethereum/core/tracing"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/trie/utils"
+	"github.com/holiman/uint256"
+)
+
+type stateDBLogger struct {
```

Comment:
I think you can define the stateDbLogger like

```go
type stateDBLogger struct {
    *StateDB
   logger *tracing.Hooks
}
```

So the stateDBlogger will inherit the functions from stateDb and you can overwrite the function you want by defining it again.

Reply:
That syntax is a bit black magic to me. If I do that, how do I then invoke the inner / parent method? E.g. to do 
```
func (s *stateDBLogger) SetNonce(address common.Address, nonce uint64) {
	s.inner.SetNonce(address, nonce)
	if s.logger != nil && s.logger.OnNonceChange != nil {
		s.logger.OnNonceChange(address, nonce-1, nonce)
	}
}
```

?

Reply:
Good idea, that made it a lot smaller!